sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/TypeDeclarationSyntaxExtensions.cs

```diff
@@ -0,0 +1,46 @@
+/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2019 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Helpers.Wrappers;
+using SonarAnalyzer.ShimLayer.CSharp;
+
+namespace SonarAnalyzer.Helpers
+{
+    public static class TypeDeclarationSyntaxExtensions
+    {
+        /// <summary>
+        /// Returns a union of all the methods and local functions from a given type declaration.
+        /// </summary>
+        public static IEnumerable<IMethodDeclaration> GetMethodDeclarations(this TypeDeclarationSyntax typeDeclaration) =>
```

Comment:
It might me overkill, but It would be nice to have unit test ensuring this new method helper logic works as intended